flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1064,6 +1076,13 @@ public void handleError(final Exception exception) {
 	 */
 	protected abstract void initialize() throws ResourceManagerException;
 
+	/**
+	 * Terminates the framework specific components.
+	 *
+	 * @throws Throwable with occurs during termination.
+	 */
+	protected abstract void terminate() throws Throwable;
```

Comment:
Can we weaken the signature to `throws Exception`?

Reply:
I kept it `Throwable` because this was what `KubernetesResourceManager#onStop` and `YarnResourceManager#onStop` catch.
Weaken to throwing `Exception` also makes sense to me.